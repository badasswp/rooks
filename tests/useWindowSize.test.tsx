import React from 'react';
import '@testing-library/jest-dom';
import { getByTestId, render } from '@testing-library/react';

import WindowSize from '../hooks/useWindowSize/WindowSize';

describe('<UseWindowSize />', () => {
  it('should render the component with no issues', () => {
    const { container } = render(<WindowSize />);

    expect(container).toMatchSnapshot();
  });

  it('should display the width & height using the useWindowSize hook', () => {
    Object.defineProperty(window, 'innerWidth', {
      value: 1366,
      writable: true,
    });

    Object.defineProperty(window, 'innerHeight', {
      value: 768,
      writable: true,
    });

    const { container } = render(<WindowSize />);

    expect(getByTestId(container, 'width').textContent).toBe('Your current Window width is 1366');

    expect(getByTestId(container, 'height').textContent).toBe('Your current Window height is 768');
  });
});
