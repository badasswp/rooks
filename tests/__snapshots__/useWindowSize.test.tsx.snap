// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseWindowSize /> should render the component with no issues 1`] = `
<div>
  <div>
    <p>
      Try to resize your Window
    </p>
    <p
      data-testid="width"
    >
      Your current Window width is 
      1024
    </p>
    <p
      data-testid="height"
    >
      Your current Window height is 
      768
    </p>
  </div>
</div>
`;
